apiserver/facades/client/spaces/spaces.go

```diff
@@ -304,17 +304,14 @@ func (api *API) ReloadSpaces(ctx stdcontext.Context) error {
 // checkSupportsSpaces checks if the environment implements NetworkingEnviron
 // and also if it supports spaces. If we don't support spaces, an
 // [errors.NotSupported] error will be returned.
-func (api *API) checkSupportsSpaces(ctx stdcontext.Context) error {
-	env, err := api.providerGetter(ctx)
-	if err != nil {
-		return fmt.Errorf("getting networking environ for model: %w", err)
-	}
+func (api *API) checkSupportsSpaces(ctx context.Context) error {
 	invalidatorFunc, err := api.credentialInvalidatorGetter()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	callCtx := envcontext.WithCredentialInvalidator(ctx, invalidatorFunc)
-	if !environs.SupportsSpaces(callCtx, env) {
+	if supported, err := api.networkService.SupportsSpaces(ctx, invalidatorFunc); err != nil {
```

Comment:
I would have just declared it without the invalidator, which can literally never be called. Save yourself getting in the first place.

Reply:
If that's the case, I suggest dropping it completely from the interface. I'll double-check here.

Reply:
This is indeed the case, we should drop the invalidator completely for this method.